### ISSUE TITLE
3.10.1: documentation build fixes and sphinx warnings

### ISSUE THREAD

**Issue by kloczek** [Legrandin/pycryptodome#546]
Looks like quite long time sphinx documentation has not been build :)

Forst I found that setuptsools<>sphinx integration is not finished.
```console
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
Testing support for clang
Target does not support clang
Testing support for gcc
Target does support gcc
Testing support for stdint.h header
Target does support stdint.h header
Testing support for 128-bit integer
Target does support 128-bit integer
Testing support for cpuid.h header
Target does support cpuid.h header
Testing support for intrin.h header
Target does not support intrin.h header
Testing support for posix_memalign
Target does support posix_memalign
Testing support for SSE2(intrin.h)
Target does not support SSE2(intrin.h)
Testing support for SSE2(x86intrin.h)
Target does support SSE2(x86intrin.h)
Testing support for AESNI intrinsics
Target does support AESNI intrinsics
Compiling support for AESNI instructions
Testing support for CLMUL intrinsics
Target does support CLMUL intrinsics
Compiling support for CLMUL instructions
running build_sphinx
Running Sphinx v4.1.2

Configuration error:
config directory doesn't contain a conf.py file (/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1)
error: Bad exit status from /var/tmp/rpm-tmp.eFyqma (%build)

```
That integration uses by default `docs/` directory so if other directory is used small adjustment in setup.cfg needs to be added:
```patch
--- a/setup.cfg~        2021-02-08 21:14:54.000000000 +0000
+++ b/setup.cfg 2021-09-19 15:20:36.603640524 +0100
@@ -7,3 +7,6 @@

 [bdist_wheel]
 py-limited-api = cp35
+
+[build_sphinx]
+source-dir = Doc
```
Second ... looks like Doc/copy.py file still has some python 3.x syntax.
```console
[tkloczko@barrel pycryptodome-3.10.1]$ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
Testing support for clang
Target does not support clang
Testing support for gcc
Target does support gcc
Testing support for stdint.h header
Target does support stdint.h header
Testing support for 128-bit integer
Target does support 128-bit integer
Testing support for cpuid.h header
Target does support cpuid.h header
Testing support for intrin.h header
Target does not support intrin.h header
Testing support for posix_memalign
Target does support posix_memalign
Testing support for SSE2(intrin.h)
Target does not support SSE2(intrin.h)
Testing support for SSE2(x86intrin.h)
Target does support SSE2(x86intrin.h)
Testing support for AESNI intrinsics
Target does support AESNI intrinsics
Compiling support for AESNI instructions
Testing support for CLMUL intrinsics
Target does support CLMUL intrinsics
Compiling support for CLMUL instructions
running build_sphinx
Running Sphinx v4.1.2

Configuration error:
There is a syntax error in your configuration file: Missing parentheses in call to 'print'. Did you mean print(sys.path)? (conf.py, line 18)
```
So here is the patch generated using `2to3` script:
```patch
--- a/Doc/conf.py       (original)
+++ b/Doc/conf.py       (refactored)
@@ -15,7 +15,7 @@

 # Modules to document with autodoc are in another directory
 sys.path.insert(0, os.path.abspath('../lib'))
-print sys.path
+print(sys.path)

 # Mock existence of native modules
 from Crypto.Util import _raw_api
@@ -60,8 +60,8 @@
 master_doc = 'index'

 # General information about the project.
-project = u'PyCryptodome'
-copyright = u'2017, Helder Eijs'
+project = 'PyCryptodome'
+copyright = '2017, Helder Eijs'

 # The version info for the project you're documenting, acts as replacement for
 # |version| and |release|, also used in various other places throughout the
@@ -224,8 +224,8 @@
 # Grouping the document tree into LaTeX files. List of tuples
 # (source start file, target name, title, author, documentclass [howto/manual]).
 latex_documents = [
-  ('index', 'PyCryptodome.tex', u'PyCryptodome Documentation',
-   u'Legrandin', 'manual'),
+  ('index', 'PyCryptodome.tex', 'PyCryptodome Documentation',
+   'Legrandin', 'manual'),
 ]

 # The name of an image file (relative to this directory) to place at the top of
@@ -254,8 +254,8 @@
 # One entry per manual page. List of tuples
 # (source start file, name, description, authors, manual section).
 man_pages = [
-    ('index', 'pycryptodome', u'PyCryptodome Documentation',
-     [u'Legrandin'], 1)
+    ('index', 'pycryptodome', 'PyCryptodome Documentation',
+     ['Legrandin'], 1)
 ]

 # If true, show URL addresses after external links.
@@ -268,8 +268,8 @@
 # (source start file, target name, title, author,
 #  dir menu entry, description, category)
 texinfo_documents = [
-  ('index', 'PyCryptodome', u'PyCryptodome Documentation',
-   u'Legrandin', 'PyCryptodome', 'One line description of project.',
+  ('index', 'PyCryptodome', 'PyCryptodome Documentation',
+   'Legrandin', 'PyCryptodome', 'One line description of project.',
    'Miscellaneous'),
 ]
```
Than I found that copy.py still uses sphionx 1.x when latest version is 4.1.2.
```console
[tkloczko@barrel pycryptodome-3.10.1]$ rpmbuild -ba --with check --with failing_tests python-pycryptodome.spec
error: failed to stat /home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/python-pycryptodome.spec: No such file or directory
[tkloczko@barrel pycryptodome-3.10.1]$ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
Testing support for clang
Target does not support clang
Testing support for gcc
Target does support gcc
Testing support for stdint.h header
Target does support stdint.h header
Testing support for 128-bit integer
Target does support 128-bit integer
Testing support for cpuid.h header
Target does support cpuid.h header
Testing support for intrin.h header
Target does not support intrin.h header
Testing support for posix_memalign
Target does support posix_memalign
Testing support for SSE2(intrin.h)
Target does not support SSE2(intrin.h)
Testing support for SSE2(x86intrin.h)
Target does support SSE2(x86intrin.h)
Testing support for AESNI intrinsics
Target does support AESNI intrinsics
Compiling support for AESNI instructions
Testing support for CLMUL intrinsics
Target does support CLMUL intrinsics
Compiling support for CLMUL instructions
running build_sphinx
Running Sphinx v4.1.2
['/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/lib', '/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1', '/usr/lib64/python38.zip', '/usr/lib64/python3.8', '/usr/lib64/python3.8/lib-dynload', '/home/tkloczko/.local/lib/python3.8/site-packages', '/usr/lib64/python3.8/site-packages', '/usr/lib/python3.8/site-packages']
making output directory... done

Exception occurred:
  File "/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/conf.py", line 158, in setup
    app.add_stylesheet('theme_overrides.css')
AttributeError: 'Sphinx' object has no attribute 'add_stylesheet'
The full traceback has been saved in /tmp/sphinx-err-l32xmsrc.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
```
So here is fix for that
```patch
--- a/Doc/conf.py~      2021-09-19 15:24:43.949253920 +0100
+++ b/Doc/conf.py       2021-09-19 15:27:47.217706142 +0100
@@ -155,7 +155,7 @@
 # Ensure that text wrapping works in a table, by overring some CSS.
 # See https://github.com/rtfd/sphinx_rtd_theme/issues/117
 def setup(app):
-    app.add_stylesheet('theme_overrides.css')
+    app.add_css_file('theme_overrides.css')

 # If not '', a 'Last updated on:' timestamp is inserted at every page bottom,
 # using the given strftime format.
```
Than at the and was possible to generate documentation however there are still some warnings to fix:
```console
[tkloczko@barrel pycryptodome-3.10.1]$ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
Testing support for clang
Target does not support clang
Testing support for gcc
Target does support gcc
Testing support for stdint.h header
Target does support stdint.h header
Testing support for 128-bit integer
Target does support 128-bit integer
Testing support for cpuid.h header
Target does support cpuid.h header
Testing support for intrin.h header
Target does not support intrin.h header
Testing support for posix_memalign
Target does support posix_memalign
Testing support for SSE2(intrin.h)
Target does not support SSE2(intrin.h)
Testing support for SSE2(x86intrin.h)
Target does support SSE2(x86intrin.h)
Testing support for AESNI intrinsics
Target does support AESNI intrinsics
Compiling support for AESNI instructions
Testing support for CLMUL intrinsics
Target does support CLMUL intrinsics
Compiling support for CLMUL instructions
running build_sphinx
Running Sphinx v4.1.2
['/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/lib', '/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1', '/usr/lib64/python38.zip', '/usr/lib64/python3.8', '/usr/lib64/python3.8/lib-dynload', '/home/tkloczko/.local/lib/python3.8/site-packages', '/usr/lib64/python3.8/site-packages', '/usr/lib/python3.8/site-packages']
making output directory... done
building [mo]: targets for 0 po files that are out of date
building [man]: all manpages
updating environment: [new config] 67 added, 0 changed, 0 removed
reading sources... [100%] src/vs_pycrypto
looking for now-outdated files... none found
pickling environment... done
checking consistency... /home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/aes.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/arc2.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/arc4.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/blowfish.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/cast.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/chacha20.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/chacha20_poly1305.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/des.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/des3.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/oaep.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/pkcs1_v1_5.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/cipher/salsa20.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/blake2b.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/blake2s.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/cmac.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/hmac.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/keccak.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/md2.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/md5.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/poly1305.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/ripemd160.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha1.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha224.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha256.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha384.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha3_224.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha3_256.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha3_384.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha3_512.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/sha512.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/shake128.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/hash/shake256.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/public_key.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/signature/dsa.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/signature/pkcs1_pss.rst: WARNING: document isn't included in any toctree
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/Doc/src/signature/pkcs1_v1_5.rst: WARNING: document isn't included in any toctree
done
writing... pycryptodome.1 { src/introduction src/features src/installation src/vs_pycrypto src/api src/cipher/cipher src/cipher/classic src/cipher/modern src/signature/signature src/hash/hash src/public_key/public_key src/public_key/rsa src/public_key/dsa src/public_key/ecc src/public_key/elgamal src/protocol/protocol src/protocol/kdf src/protocol/ss src/io/io src/io/pem src/io/pkcs8 src/random/random src/util/util src/util/asn1 src/examples src/faq src/contribute_support src/future src/changelog src/license } /home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/lib/Crypto/PublicKey/ECC.py:docstring of Crypto.PublicKey.ECC.EccKey:5: WARNING: Failed to create a cross reference. Any number is not assigned: curve-names
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/lib/Crypto/PublicKey/ECC.py:docstring of Crypto.PublicKey.ECC.construct:4: WARNING: Failed to create a cross reference. Any number is not assigned: curve-names
/home/tkloczko/rpmbuild/BUILD/pycryptodome-3.10.1/lib/Crypto/PublicKey/ECC.py:docstring of Crypto.PublicKey.ECC.generate:3: WARNING: Failed to create a cross reference. Any number is not assigned: curve-names
done
build succeeded, 39 warnings.

The manual pages are in build/sphinx/man.
```
Please let me know if you want above fixes as PR. Otherwise feel free to commit those chenges without PR.

**Comment by kloczek** [comment]
Alternatively instead first patch of the setup.cfg Doc/ needs to be renamed to docs/ :)

**Comment by Legrandin** [comment]
Thanks for pointing this out.
Docs are built by Readthedocs, which is still configured to use Python 2.7.
I will move it over to Python 3.

**Comment by kloczek** [comment]
Pleae let me know if you will have anything to test :)
Thx.

**Comment by Legrandin** [comment]
Fixed with 0b25c473bca872e08e991f0dcca0d34ce7dd8641, 0b25c473bca872e08e991f0dcca0d34ce7dd8641, 0b25c473bca872e08e991f0dcca0d34ce7dd8641

**Comment by kloczek** [comment]
> Fixed with [0b25c47](https://github.com/Legrandin/pycryptodome/commit/0b25c473bca872e08e991f0dcca0d34ce7dd8641), [0b25c47](https://github.com/Legrandin/pycryptodome/commit/0b25c473bca872e08e991f0dcca0d34ce7dd8641), [0b25c47](https://github.com/Legrandin/pycryptodome/commit/0b25c473bca872e08e991f0dcca0d34ce7dd8641)

Three times the same hash ..
Hmm I don't see this commit on master and when I'm trying to download https://github.com/Legrandin/pycryptodome/commit/0b25c47.patch I have `curl: (22) The requested URL returned error: 403`
